addons/account/models/account_move.py

```diff
@@ -2987,15 +2987,16 @@ def action_switch_invoice_into_refund_credit_note(self):
                 'partner_bank_id': False,
                 'currency_id': move.currency_id.id,
             })
-            move.write({
-                'line_ids': [
-                    Command.update(line.id, {
-                        'quantity': -line.quantity,
-                    })
-                    for line in move.line_ids
-                    if line.display_type == 'product' and line.quantity < 0
-                ]
-            })
+            if move.amount_total < 0:
```

Comment:
```suggestion
            if move.amount_total >= 0:
                continue
```
to avoid an extra indent? It's just a suggestion. Ignore it if you don't agree